File: src/com/sterndu/util/ReturnType.java
package com.sterndu.util;

import java.util.Collection;
import java.util.List;
import java.util.Set;

public class ReturnType<E> {

	private final E retval;

	public ReturnType(E val) {
		retval = val;
	}

	public E getVal() { return retval; }

	public boolean isArray() { return retval.getClass().isArray(); }

	public boolean isCollection() { return Collection.class.isInstance(retval); }

	public boolean isInstanceOf(Class<?> clazz) { return clazz.isInstance(retval); }

	public boolean isList() { return List.class.isInstance(retval); }

	public boolean isMultipleVals() { return isCollection() | isArray(); }

	public boolean isSet() {
		return Set.class.isInstance(retval);
	}

}

File: src/com/sterndu/util/ReturnType.kt
@file:JvmName("ReturnType")
package com.sterndu.util

class ReturnType<out E : Any>(private val `val`: E) {

	val isArray: Boolean
		get() = `val`.javaClass.isArray
	val isCollection: Boolean
		get() = `val` is Collection<*>

	fun isInstanceOf(clazz: Class<*>): Boolean {
		return clazz.isInstance(`val`)
	}

	val isList: Boolean
		get() = `val` is List<*>
	val isMultipleVals: Boolean
		get() = isCollection or isArray
	val isSet: Boolean
		get() = `val` is Set<*>
}
